syntax = "proto3";

package Messages;

message Response {
  message Stop {
    string name = 1;
    repeated string bus_names = 2;
  }
  message Bus {
    string name = 1;
    uint32 stop_count = 2;
    uint32 unique_stop_count = 3;
    int32 road_route_length = 4;
    double geo_route_length = 5;
  }
}

message TransportRouter {
  message RoutingSettings {
    int32 bus_wait_time = 1;
    double bus_velocity = 2;
  }
  message StopVertexIds {
    string name = 1;
    uint64 in = 2;
    uint64 out = 3;
  }
  message VertexInfo {
    string stop_name = 1;
  }
  message BusEdgeInfo {
    string bus_name = 1;
    uint64 start_stop_idx = 2;
    uint64 finish_stop_idx = 3;
    uint64 start_vertex_id = 4;
    uint64 finish_vertex_id = 5;
    double ride_time = 6;
  }

  RoutingSettings routing_settings = 1;
  repeated VertexInfo vertices_info = 2;
  repeated StopVertexIds stops_vertex_ids = 3;
  repeated BusEdgeInfo bus_edges_info = 4;
}

message TransportCatalog {
  repeated Response.Stop stops_responses = 1;
  repeated Response.Bus buses_responses = 2;
  TransportRouter transport_router = 3;
}